src/wp-admin/includes/widgets.php

```diff
@@ -242,17 +259,22 @@ function wp_widget_control( $sidebar_args ) {
 	if ( isset( $control['callback'] ) ) {
 		$has_form = call_user_func_array( $control['callback'], $control['params'] );
 	} else {
-		echo "\t\t<p>" . __('There are no options for this widget.') . "</p>\n";
+		echo "\t\t<p>" . __( 'There are no options for this widget.' ) . "</p>\n";
+	}
+
+	$noform_class = '';
+	if ( 'noform' === $has_form ) {
+		$noform_class = ' widget-control-noform';
 	}
 	?>
 	<?php echo $after_widget_content; ?>
-	<input type="hidden" name="widget-id" class="widget-id" value="<?php echo esc_attr($id_format); ?>" />
-	<input type="hidden" name="id_base" class="id_base" value="<?php echo esc_attr($id_base); ?>" />
-	<input type="hidden" name="widget-width" class="widget-width" value="<?php if (isset( $control['width'] )) echo esc_attr($control['width']); ?>" />
-	<input type="hidden" name="widget-height" class="widget-height" value="<?php if (isset( $control['height'] )) echo esc_attr($control['height']); ?>" />
-	<input type="hidden" name="widget_number" class="widget_number" value="<?php echo esc_attr($widget_number); ?>" />
-	<input type="hidden" name="multi_number" class="multi_number" value="<?php echo esc_attr($multi_number); ?>" />
-	<input type="hidden" name="add_new" class="add_new" value="<?php echo esc_attr($add_new); ?>" />
+	<input type="hidden" name="widget-id" class="widget-id" value="<?php echo esc_attr( $id_format ); ?>" />
+	<input type="hidden" name="id_base" class="id_base" value="<?php echo esc_attr( $id_base ); ?>" />
+	<input type="hidden" name="widget-width" class="widget-width" value="<?php echo esc_attr( $width ); ?>" />
```

Comment:
This is much less legible. The variable is set way up on line 197. I don't like this one (or next line).

Reply:
I think variables like `$id_format` and `$od_base` were always set much higher in this file, if that's what you mean.

Reply:
No, my comment was for the width (and height). I don't even remember seeing the other two. It just makes it more difficult to maintain.

Reply:
Updated PR to follow.